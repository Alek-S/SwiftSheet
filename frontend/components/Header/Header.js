import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import defaultStyle from '../../defaultStyle';
import { Link, NavLink } from 'react-router-dom';

/**
 * @function
 * @name Header
 * Top header element of page - always viewable
 *
 * @returns {JSX}
 */
const Header = () => {
	const [headerSize, setHeaderSize] = useState('large');

	useEffect(() => {
		window.addEventListener('scroll', _handleScroll);

		return () => {
			window.removeEventListener('scroll', _handleScroll);
		};
	});

	/**
	 * @function
	 * Updates headerSize state
	 * depending on how far scrolled from top
	 */
	const _handleScroll = () => {
		const scrollTop = document.documentElement.scrollTop;
		const resize = 50; //distance from top in px at which to resize header

		if (scrollTop > resize && headerSize !== 'small') {
			setHeaderSize('small');
		} else if (scrollTop <= resize && headerSize !== 'large') {
			setHeaderSize('large');
		}
	};

	return (
		<Nav className={headerSize}>
			<Link to="/">
				<img src="./assets/images/logo.svg" alt="logo" /> <h1>SwiftSheet</h1>
			</Link>
			<Tagline className="tagline">
				<h2>Fast and ephemeral data sharing.</h2>
			</Tagline>

			<NavSection className={headerSize}>
				<StyledNavLink to="/upload">Upload</StyledNavLink>
				<StyledNavLink to="/view">View</StyledNavLink>
			</NavSection>
		</Nav>
	);
};

const Nav = styled(defaultStyle)`
	position: fixed;
	border-bottom: solid 1px ${props => props.theme.color.border};
	padding: 0.8rem;
	transition: all 0.3s;
	width: 100%;
	z-index: 50;

	a {
		text-decoration: none;
	}

	& img,
	h1 {
		display: inline;
		transition: all 0.3s;
	}
	& img {
		height: 50px;
	}
	& h1 {
		font-family: ${props => props.theme.font.header};
		font-size: 2.5rem;
		letter-spacing: 4px;
		margin-left: 0.5rem;
		position: relative;
		top: -0.7rem;
		color: ${props => props.theme.color.text};
	}

	&.small {
		padding: 0.4rem;
		padding-left: 0.8rem;
		padding-bottom: 0.15rem;

		img {
			height: 25px;
		}
		h1 {
			font-size: 1.5rem;
			top: -0.3rem;
		}
		.tagline {
			display: none;
		}
	}

	@media (max-width: 475px) {
		h1 {
			top: -0.3rem;
			font-size: 1.25rem;
		}
		img {
			height: 25px;
		}
	}

	@media (max-width: 350px) {
		h1 {
			top: 0rem;
			font-size: 1.15rem;
		}
		img {
			position: relative;
			height: 20px;
			top: 0.25rem;
		}
	}
`;

const NavSection = styled.ul`
	position: absolute;
	top: 1.75rem;
	right: 2rem;
	transition: all 0.3s;
	&.small {
		top: 0.6rem;
	}
	&.small * {
		font-size: 1rem;
	}

	@media (max-width: 475px) {
		top: 1.2rem;
	}
`;

/* istanbul ignore next */
const StyledNavLink = styled(NavLink)`
	font-size: 1.25rem;
	color: ${props => props.theme.color.text};
	padding-right: 2rem;
	padding-left: 2rem;
	transition: all 0.3s;
	border-left: solid 1px ${props => props.theme.color.border};

	&.active {
		font-weight: 500;
		color: ${props => props.theme.color.red};
	}

	@media (max-width: 550px) {
		& {
			font-size: 1.1rem;
			padding-right: 1rem;
			padding-left: 1rem;
		}
	}

	@media (max-width: 350px) {
		padding-right: 0.5rem;
		padding-left: 0.5rem;
	}
`;

const Tagline = styled.div`
	color: ${props => props.theme.color.lightText};
	position: absolute;
	display: inline-block;
	margin-left: 2rem;
	margin-top: 0.5rem;
	padding: 0.75rem 2rem;
	border-left: solid 1px ${props => props.theme.color.border};

	@media (max-width: 835px) {
		& {
			display: none;
		}
	}
`;

export default Header;
export { Nav, StyledNavLink, Tagline };
