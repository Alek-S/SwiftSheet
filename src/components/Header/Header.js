import React, { Component } from 'react';
import styled from 'styled-components';
import defaultStyle from '../../defaultStyle';

/** @class
 * @name Header
 * Top header element of page - always viewable
 *
 * @returns {JSX}
 */
export default class Header extends Component {
	constructor(props) {
		super(props);

		this.state = {
			headerSize: 'large',
		};
	}
	componentDidMount() {
		window.addEventListener('scroll', this.handleScroll);
	}

	/**
	 * Updates this.state.headerSize depending on how far
	 * scrolled from top
	 *
	 * @param {Object} Event
	 * @param {Number} scroll top override value otherwise based on document's scrollTop
	 */
	handleScroll = (e, scrollTop = document.documentElement.scrollTop) => {
		//distance from top in px at which to resize header
		const resize = 50;

		if (scrollTop > resize && this.state.headerSize !== 'small') {
			this.setState({
				headerSize: 'small',
			});
		} else if (scrollTop <= resize && this.state.headerSize !== 'large') {
			this.setState({
				headerSize: 'large',
			});
		}
	};

	render() {
		const { headerSize } = this.state;

		return (
			<Nav className={headerSize}>
				<img src="./assets/images/logo.svg" alt="logo" /> <h1>SwiftSheet</h1>
			</Nav>
		);
	}
}

const Nav = styled(defaultStyle)`
	border-bottom: solid 1px ${props => props.theme.color.border};
	padding: 0.8rem;
	position: fixed;
	transition: all 0.3s;
	width: 100%;

	& img,
	h1 {
		display: inline;
		transition: all 0.3s;
	}
	& img {
		height: 50px;
	}
	& h1 {
		font-family: ${props => props.theme.font.header};
		font-size: 2.5rem;
		letter-spacing: 4px;
		margin-left: 0.5rem;
		position: relative;
		top: -0.7rem;
	}

	&.small {
		padding: 0.4rem;
		padding-left: 0.8rem;
		padding-bottom: 0.15rem;

		img {
			height: 25px;
		}
		h1 {
			font-size: 1.5rem;
			top: -0.3rem;
		}
	}
`;
