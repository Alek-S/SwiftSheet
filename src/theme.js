/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
