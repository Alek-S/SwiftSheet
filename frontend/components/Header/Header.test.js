import React from 'react';
import 'jest-styled-components';
import { BrowserRouter as Router } from 'react-router-dom';

import Header from './Header';
import theme from '../../theme';

// simulate window scroll
function fireScroll(height) {
	document.documentElement.scrollTop = height;
	window.dispatchEvent(new Event('scroll'));
}

describe('Header', () => {
	test('matches the snapshot', () => {
		const tree = mountWithTheme(
			<Router>
				<Header />
			</Router>,
			theme
		);
		expect(toJson(tree)).toMatchSnapshot();
	});

	test('start as large', () => {
		const wrapper = shallow(<Header />);
		expect(wrapper.find('.large').length).toBeGreaterThan(0);
	});
});
