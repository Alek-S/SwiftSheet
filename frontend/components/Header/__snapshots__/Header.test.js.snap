// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
.c1 {
  background-color: white;
  color: #404040;
  font-family: 'Quicksand',sans-serif;
  font-size: 0.9em;
}

.c0 {
  position: fixed;
  border-bottom: solid 1px #d9d9d9;
  padding: 0.8rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  width: 100%;
  z-index: 50;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 img,
.c0 h1 {
  display: inline;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 img {
  height: 50px;
}

.c0 h1 {
  font-family: 'Poiret One',cursive;
  font-size: 2.5rem;
  -webkit-letter-spacing: 4px;
  -moz-letter-spacing: 4px;
  -ms-letter-spacing: 4px;
  letter-spacing: 4px;
  margin-left: 0.5rem;
  position: relative;
  top: -0.7rem;
  color: #404040;
}

.c0.small {
  padding: 0.4rem;
  padding-left: 0.8rem;
  padding-bottom: 0.15rem;
}

.c0.small img {
  height: 25px;
}

.c0.small h1 {
  font-size: 1.5rem;
  top: -0.3rem;
}

.c0.small .tagline {
  display: none;
}

.c3 {
  position: absolute;
  top: 1.75rem;
  right: 2rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c3.small {
  top: 0.6rem;
}

.c3.small * {
  font-size: 1rem;
}

.c4 {
  font-size: 1.25rem;
  color: #404040;
  padding-right: 2rem;
  padding-left: 2rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  border-left: solid 1px #d9d9d9;
}

.c4.active {
  font-weight: 500;
  color: #ED5050;
}

.c2 {
  color: #848484;
  position: absolute;
  display: inline-block;
  margin-left: 2rem;
  margin-top: 0.5rem;
  padding: 0.75rem 2rem;
  border-left: solid 1px #d9d9d9;
}

@media (max-width:475px) {
  .c0 h1 {
    top: -0.3rem;
    font-size: 1.25rem;
  }

  .c0 img {
    height: 25px;
  }
}

@media (max-width:350px) {
  .c0 h1 {
    top: 0rem;
    font-size: 1.15rem;
  }

  .c0 img {
    position: relative;
    height: 20px;
    top: 0.25rem;
  }
}

@media (max-width:475px) {
  .c3 {
    top: 1.2rem;
  }
}

@media (max-width:550px) {
  .c4 {
    font-size: 1.1rem;
    padding-right: 1rem;
    padding-left: 1rem;
  }
}

@media (max-width:350px) {
  .c4 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media (max-width:835px) {
  .c2 {
    display: none;
  }
}

<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header>
      <Header__Nav
        className="large"
      >
        <defaultStyle
          className="large c0"
        >
          <div
            className="large c0 c1"
          >
            <Link
              replace={false}
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <img
                  alt="logo"
                  src="./assets/images/logo.svg"
                />
                 
                <h1>
                  SwiftSheet
                </h1>
              </a>
            </Link>
            <Header__Tagline
              className="tagline"
            >
              <div
                className="tagline c2"
              >
                <h2>
                  Fast and ephemeral data sharing.
                </h2>
              </div>
            </Header__Tagline>
            <Header__NavSection
              className="large"
            >
              <ul
                className="large c3"
              >
                <Header__StyledNavLink
                  to="/upload"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="c4"
                    to="/upload"
                  >
                    <Route
                      path="\\\\/upload"
                    >
                      <Link
                        aria-current={null}
                        className="c4"
                        replace={false}
                        to="/upload"
                      >
                        <a
                          aria-current={null}
                          className="c4"
                          href="/upload"
                          onClick={[Function]}
                        >
                          Upload
                        </a>
                      </Link>
                    </Route>
                  </NavLink>
                </Header__StyledNavLink>
                <Header__StyledNavLink
                  to="/view"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="c4"
                    to="/view"
                  >
                    <Route
                      path="\\\\/view"
                    >
                      <Link
                        aria-current={null}
                        className="c4"
                        replace={false}
                        to="/view"
                      >
                        <a
                          aria-current={null}
                          className="c4"
                          href="/view"
                          onClick={[Function]}
                        >
                          View
                        </a>
                      </Link>
                    </Route>
                  </NavLink>
                </Header__StyledNavLink>
              </ul>
            </Header__NavSection>
          </div>
        </defaultStyle>
      </Header__Nav>
    </Header>
  </Router>
</BrowserRouter>
`;
